### PR TITLE
File sync example

### PR DESCRIPTION
**Pull request by jacobythwaites**
Note that you can put this processing instruction after the XML declaration in source files to have the rawgit.com-served XML render as HTML:

```
<?xml-stylesheet type="text/xsl" href="/opensparkl/sse_cli/master/sparkl_cli/resources/render.xsl"?>
```

### CODE REVIEW

File: Examples/FileSync/master.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  """
+  def __init__(self, service):
+    self.service = service
+
+  def on_created(self, event):
+    binary = bytearray()
+    if not event.is_directory:
+      with open(event.src_path, mode='rb') as file:
+        binary = file.read()
+
```
Comment (miklosduma) [inline Examples/FileSync/master.py]: Extra blank line

File: Examples/FileSync/master.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    put(self.service, event.is_directory, event.src_path, binary)
+
+  def on_deleted(self, event):
+      delete(self.service, event.is_directory, event.src_path)
+
+  def on_modified(self, event):
```
Comment (miklosduma) [inline Examples/FileSync/master.py]: on_modified and on_created looks nearly identical

File: Examples/TinyTableRest/index.html [hunk 3]
```diff
@@ -1,63 +0,0 @@
-<!DOCTYPE html>
```
Comment (miklosduma) [inline Examples/TinyTableRest/index.html]: Please restore :)

File: Examples/FileSync/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+Middleware mix (click to view, right-click to save):
+
+* [FileSync](https://rawgit.com/jacobythwaites/examples/file_sync/Examples/FileSync/FileSync.xml)
+
+Video of this example (4 minutes):
+
+* [FileSync video](https://vimeo.com/289874337)
+
+## Dependencies
```
[the remark this answers is not in the document]
Comment (jacobythwaites) [reply Examples/FileSync/README.md]: Done

File: Examples/FileSync/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+* Use `git clone https://opensparkl/examples` to clone this repo
+* The `master.py` agent uses the [watchdog](https://pythonhosted.org/watchdog/) utility which
+  must be installed first.
+
+## Instructions
+1. Change to the `Examples/FileSync` directory and save it as $PKG_DIR
+```
```
[the remark this answers is not in the document]
Comment (jacobythwaites) [reply Examples/FileSync/README.md]: Done

File: Examples/FileSync/master.py (hunk 2 above)
Comment (jacobythwaites) [reply Examples/FileSync/master.py]: Almost. But not quite ;)

File: Examples/TinyTableRest/index.html (hunk 3 above)
Comment (jacobythwaites) [reply Examples/TinyTableRest/index.html]: Um, I think that was a .gitignore entry put in by a certain miklosduma

**Pull request merged**